src/web/server/app/tests/controllers/api.test.ts

```diff
@@ -0,0 +1,262 @@
+      jest
+        .spyOn(apiServices, 'verifyUserCamera')
+        .mockImplementationOnce((userId: string, cameraId: string) =>
+          Promise.resolve(true)
+        );
+      jest
+        .spyOn(apiServices, 'retrieveStreamIds')
+        .mockImplementationOnce((cameraId: string) =>
+          Promise.resolve(undefined as any)
+        );
+      const req = mockReq(testCameraId, testUserId);
+      const res = mockRes();
+      await api.retrieveStreamIds(req, res);
+      expect(res.status).toBeCalledWith(400);
+      expect(res.send).toBeCalledWith('Invalid cameraId');
+    });
+  });
+  describe('retrieveLiveStreamId', () => {
+    const mockReq: any = (cameraId: string) => {
+      return {
+        params: {
+          cameraId,
+        },
+      };
+    };
+    const mockRes: any = () => {
+      const res = {
+        status: jest.fn(),
+        send: jest.fn(),
+        json: jest.fn(),
+      };
+      res.status = jest.fn().mockReturnValue(res);
+      res.send = jest.fn().mockReturnValue(res);
+      res.json = jest.fn().mockReturnValue(res);
+      return res;
+    };
+    const testCameraId = 'randomCameraId';
+    const testStreamId = 'randomStreamId';
+    it('should return testStreamIds list', async () => {
+      const mockResults = { rows: [{ streamId: testStreamId }] };
+      jest
+        .spyOn(apiServices, 'retrieveLiveStreamId')
+        .mockImplementationOnce((cameraId: string) =>
+          Promise.resolve(mockResults as any)
+        );
+      const req = mockReq(testCameraId);
+      const res = mockRes();
+      await api.retrieveLiveStreamId(req, res);
+      expect(res.status).toBeCalledWith(200);
+      expect(res.json).toBeCalledWith(testStreamId);
+    });
+    it('should return 400 status if user can not view', async () => {
+      jest
+        .spyOn(apiServices, 'retrieveLiveStreamId')
+        .mockImplementationOnce((cameraId: string) =>
+          Promise.resolve(undefined as any)
+        );
+      const req = mockReq(testCameraId);
+      const res = mockRes();
+      await api.retrieveLiveStreamId(req, res);
+      expect(res.status).toBeCalledWith(400);
+      expect(res.send).toBeCalledWith('Invalid cameraId');
+    });
+  });
+  describe('retrieveLiveCameraStreamIds', () => {
+    const mockReq: any = (userId: string) => {
+      return {
+        session: { userId },
+      };
+    };
+    const mockRes: any = () => {
+      const res = {
+        status: jest.fn(),
+        send: jest.fn(),
+        json: jest.fn(),
+      };
+      res.status = jest.fn().mockReturnValue(res);
+      res.send = jest.fn().mockReturnValue(res);
+      res.json = jest.fn().mockReturnValue(res);
+      return res;
+    };
+    const testCameraIds = ['randomCameraId1', 'randomCameraId2'];
+    const testStreamIds = ['randomStreamId1', 'randomStreamId2'];
+    it('should return a dictionary of cameraIds to streamIds', async () => {
+      const mockResults = {} as any;
```

Comment:
Isnt this type `Record<string, string>` ?

Reply:
done